### ISSUE TITLE
[docs] [community] Encryption at rest -- what's encrypted?

### ISSUE THREAD

**Issue by stevebang** [yugabyte/yugabyte-db#3260]
https://yugabyte-db.slack.com/archives/CG0KQF0GG/p1578180638319900

Naveen 3:30 PM
In the docs encryption at rest and transit are mentioned. Are log files and temporary files encrypted?

7 replies

Naveen Jan 4th at 3:30 PM
In the docs encryption at rest and transit are mentioned. Are log files and temporary files encrypted?
7 replies

Sid Choudhury  1 day ago
No they are not encrypted.

Naveen  1 day ago
Thanks, Sid.

Karthik Ranganathan  1 day ago
Just clarifying here that the user data is always encrypted (WAL, data files). The DB logs and other stuff is not.
:+1:

Naveen  1 day ago
Are encryption of DB logs and temporary files in the plan? (edited) 

Karthik Ranganathan  1 day ago
Would self encrypted SSD drives work for this purpose?

Naveen  1 day ago
I am not sure. Not an expert in this domain. I read from documents that MariaDB, SQL Server do encrypt logs. Couldn’t find info about Postgres. (edited) 

Naveen  1 day ago
Encrypted on Aurora too:
For an Amazon Aurora encrypted DB cluster, all logs, backups, and snapshots are encrypted. You can also encrypt a Read Replica of an Amazon Aurora encrypted cluster. 

Last reply 1 day agoView thread

Steve Bang 3:36 PM
Thanks @sid -- more on this topic can be found here: https://github.com/yugabyte/yugabyte-db/blob/master/architecture/design/docdb-encryption-at-rest.md#assumptions

architecture/design/docdb-encryption-at-rest.md
```
# Encryption At Rest in YugaByte DB
Data at rest within a YugaByte DB cluster should be protected from unauthorized users by encrypting it. This document outlines how this is achieved internally, along with what features we support.
## Features
Users will be able to perform the following operations:
* Enable and disable encryption on existing clusters with data.
* Backup encrypted data in efficiently. This should not involve a rewrite of the entire dataset.
* Periodically rotate encryption keys. This operation should be efficient without a rewrite of the entire dataset. Rotation of keys is done to limit the window that an attacker can use the key to access the data.
* Integrate key management with an external KMS system.
## Assumptions
This feature makes the following assumptions:
* Encryption will only be enabled for tablet server data files and WAL files. Other auxillary data such as metadata files (for example, files that store the internal uuid of a tablet server) and logs from various processes are not encrypted. These do not constitute user data.
* Attackers do not have privileged access on a running system, so they cannot read in-memory state.
* Attackers do not have write access to the encrypted raw files. Thus, there is no scheme to verify the integrity of the data in the case that an attacker corrupts it.

**Comment by stevebang** [comment]
Added to the "encryption at rest" page: a list of files that are encrypted and a separate list of files that are NOT encrypted.

**Comment by stevebang** [comment]
Fixed by PR #3334.
